pull.go

```diff
@@ -152,60 +152,60 @@ func localImageNameForReference(ctx context.Context, store storage.Store, srcRef
 
 // Pull copies the contents of the image from somewhere else to local storage.
 func Pull(ctx context.Context, imageName string, options PullOptions) error {
-	spec := imageName
 	systemContext := getSystemContext(options.SystemContext, options.SignaturePolicyPath)
-	srcRef, err := alltransports.ParseImageName(spec)
-	if err != nil {
-		if options.Transport == "" {
-			options.Transport = util.DefaultTransport
-		}
-		logrus.Debugf("error parsing image name %q, trying with transport %q: %v", spec, options.Transport, err)
-		transport := options.Transport
-		if transport != util.DefaultTransport {
-			transport = transport + ":"
-		}
-		spec = transport + spec
-		srcRef2, err2 := alltransports.ParseImageName(spec)
-		if err2 != nil {
-			return errors.Wrapf(err2, "error parsing image name %q", imageName)
-		}
-		srcRef = srcRef2
+
+	transport := options.Transport
+	if transport == "" {
+		transport = util.DefaultTransport
+	}
+	boptions := BuilderOptions{
+		FromImage:           imageName,
+		Transport:           transport,
+		SignaturePolicyPath: options.SignaturePolicyPath,
+		SystemContext:       systemContext,
+		PullBlobDirectory:   options.BlobDirectory,
 	}
+
 	if options.Quiet {
-		options.ReportWriter = nil // Turns off logging output
+		boptions.ReportWriter = nil // Turns off logging output
 	}
-	var names []string
+	srcRef, _, err := resolveImage(ctx, systemContext, options.Store, boptions)
```

Comment:
The reference returned by `resolveImage` does not refer to the source (e.g. a remote registry), but to the destination (c/storage).

_If_ the source is from a `docker.Transport`, the `DockerReference()` value should be the same, so in that case this is primarily a matter of variable naming; if the source is from some other source entirely, this code will pull the other tags from some completely unrelated docker/distribution registry, based on how the pull code decided to name the pulled image.